test/src/userobjects/ElemSideNeighborLayersVarTester.C

```diff
@@ -0,0 +1,109 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org All rights reserved, see COPYRIGHT
+//* for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT Licensed
+//* under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElemSideNeighborLayersVarTester.h"
+
+#include "libmesh/parallel.h"
+
+registerMooseObject("MooseTestApp", ElemSideNeighborLayersVarTester);
+
+template <>
+InputParameters
+validParams<ElemSideNeighborLayersVarTester>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription(
+      "Tests whether a coupled variable is correctly ghosted.  This code needs to be cleaned up");
+  params.addRequiredParam<unsigned int>("rank", "The rank for which the ghosted data are recorded");
+
+  params.registerRelationshipManagers("ElementSideNeighborLayers");
+  params.addRequiredParam<unsigned short>("element_side_neighbor_layers",
+                                          "Number of layers to ghost");
+  params.addRequiredCoupledVar("u",
+                               "The variable that is ghosted.  At nodes where this is NOT ghosted, "
+                               "this UserObject will set the variable to -1, so to visualise the "
+                               "areas where it is ghosted, initialize u to a positive value");
+  return params;
+}
+
+ElemSideNeighborLayersVarTester::ElemSideNeighborLayersVarTester(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _rank(getParam<unsigned int>("rank")),
+    _u_nodal(getVar("u", 0)),
+    _u_var_num(coupled("u", 0))
+{
+}
+
+void
+ElemSideNeighborLayersVarTester::timestepSetup()
+{
+  _nodal_data.clear();
+  auto my_processor_id = processor_id();
+
+  if (my_processor_id == _rank)
+  {
+    // here i'm looping over *all* elements for a replicated mesh and that seems a waste: only
+    // Evaluable+ghost would be better
+    for (const auto & elem : _subproblem.mesh().getMesh().active_element_ptr_range())
```

Comment:
Loop over evaluable element ranges

Reply:
No, evaluable does not loop over local+ghost.  It loops over local + 1 layer.  This is true in both this class and @permcody 's ElemSideNeighborLayersVarTester , which you can verify using the "evaluable.i" input/gold files.

Reply:
It should now!!!